domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java

```diff
@@ -171,4 +179,38 @@ private AuthorisationToAccessControlListMapper() {
                 .collect(Collectors.toList());
         }
     }
+
+    // Mapper to call fieldTypeEntity listItems only for fixedList items not for all fields.
+    // It improves performance.
+    class FixedListMapper {
+        private FixedListMapper() {
+            // Default constructor
+        }
+
+        static List<FixedListItem> map(FieldTypeEntity fieldTypeEntity) {
+            List<FixedListItem> fixedListItems = new ArrayList<>();
+            FieldTypeEntity baseFieldTypeEntity = fieldTypeEntity.getBaseFieldType();
+            if (baseFieldTypeEntity == null && FieldTypeEntity.isFixedList(fieldTypeEntity.getReference())) {
+                extractListItems(fieldTypeEntity, fixedListItems);
+            } else if (baseFieldTypeEntity != null && FieldTypeEntity.isFixedList(baseFieldTypeEntity.getReference())) {
+                extractListItems(fieldTypeEntity, fixedListItems);
+            }
+            return fixedListItems;
+        }
+
+        private static void extractListItems(final FieldTypeEntity fieldTypeEntity,
+                                             final List<FixedListItem> fixedListItems) {
+            fieldTypeEntity.getListItems().stream().forEach(fieldTypeListItemEntity ->
+                                                            {
+                                                                FixedListItem fixedListItem = new FixedListItem();
```

Comment:
can you extract this block of code into a class FixedListItemMapper which has a method `FixedListItem map(FixedListItemEntity)` and call that here please

Reply:
Happy to do, I have realised that while doing this refactor, EntityToResponseDTOMapper is a interface (mapstruct) and mapstruct using lot to generate code from annotations, not sure if we include spring dependencies here for this util extraction of code, how mapstruct will react and will it be really useful, please, suggest me.

Reply:
DONE